src/Microsoft.DotNet.Build.Tasks.Packaging/src/GetApplicableAssetsFromPackages.cs

```diff
@@ -0,0 +1,167 @@
+// See the LICENSE file in the project root for more information.
+
+using Microsoft.Build.Framework;
+using Microsoft.Build.Utilities;
+using Newtonsoft.Json;
+using Newtonsoft.Json.Linq;
+using NuGet.Frameworks;
+using System;
+using System.Collections.Generic;
+using System.Diagnostics;
+using System.IO;
+using System.Linq;
+using System.Text;
+
+namespace Microsoft.DotNet.Build.Tasks.Packaging
+{
+    public class GetApplicableAssetsFromPackages : PackagingTask
+    {
+        private Dictionary<string, List<PackageItem>> _packageToPackageItems;
+        private Dictionary<string, PackageItem> _targetPathToPackageItem;
+        private AggregateNuGetAssetResolver _resolver;
+
+        /// <summary>
+        /// All items that make up packages to resolve from.  
+        /// Must have the following metadata
+        ///     Identity - path to the file in the binary directory
+        ///         TargetPath - path to the file in the package
+        ///         PackageId - ID of the package containing the file
+        /// </summary>
+        [Required]
+        public ITaskItem[] PackageAssets { get; set; }
+
+        /// <summary>
+        /// TargetMoniker to use when resolving assets.  EG: netcoreapp1.0, netstandard1.4
+        /// </summary>
+        [Required]
+        public string TargetMoniker { get; set; }
+
+        /// <summary>
+        /// runtime.json that defines the RID graph
+        /// </summary>
+        [Required]
+        public string RuntimeFile { get; set; }
+        
+        /// <summary>
+        /// If specified will be used when resolving runtime assets, otherwise TargetMoniker will be used.
+        /// </summary>
+        public string RuntimeTargetMoniker { get; set; }
+
+        /// <summary>
+        /// If specified will be used when resolving runtime assets, otherwise no RID will be used.
+        /// </summary>
+        public string TargetRuntime { get; set; }
+
+        [Output]
+        public ITaskItem[] CompileAssets { get; set; }
+
+        [Output]
+        public ITaskItem[] RuntimeAssets { get; set; }
+
+        /// <summary>
+        /// Generates a table in markdown that lists the API version supported by 
+        /// various packages at all levels of NETStandard.
+        /// </summary>
+        /// <returns></returns>
+        public override bool Execute()
+        {
+            if (PackageAssets == null || PackageAssets.Length == 0)
+            {
+                Log.LogError("PackageAssets argument must be specified");
+                return false;
+            }
+
+            if (String.IsNullOrEmpty(TargetMoniker))
+            {
+                Log.LogError("TargetMoniker argument must be specified");
+                return false;
+            }
+
+            NuGetFramework fx = NuGetFramework.Parse(TargetMoniker);
+            NuGetFramework runtimeFx = fx;
+
+            if (!String.IsNullOrEmpty(RuntimeTargetMoniker))
```

Comment:
Yeah, suppose this is used by something other than a test.  I'm just trying to come up with the right behavior for the task, then you can decide how you want to drive it with targets.